Fails to install cryptography for `moto-1.3.15`

Inside a python:3.8-slim docker container, `pip install moto` does not install `cryptography` on the latest moto release (moto-1.3.15). This causes subsequent commands to fail with `ModuleNotFoundError: No module named 'cryptography'` when invoking `moto` mocks.

To see this, jump into a fresh docker:
`docker run --rm -it python:3.8-slim /bin/bash`

Then inside run:
```
pip install moto
pip list
```
And note that `cryptography` is not installed.

Now run:
```
pip install moto==1.3.14
pip list
```

And note that `cryptography` is now installed.

An example of the failures seen when using the new release (ie. having only run `pip install moto`):
```
/usr/local/lib/python3.8/site-packages/moto/iam/__init__.py:2: in <module>
    from .models import iam_backend
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

    from __future__ import unicode_literals
    import base64
    import hashlib
    import os
    import random
    import string
    import sys
    from datetime import datetime
    import json
    import re
    
>   from cryptography import x509
E   ModuleNotFoundError: No module named 'cryptography'

/usr/local/lib/python3.8/site-packages/moto/iam/models.py:12: ModuleNotFoundError
```

--- a/setup.py
+++ b/setup.py
@@ -99,25 +99,27 @@ all_extra_deps = [
     _dep_sshpubkeys_py2,
     _dep_sshpubkeys_py3,
 ]
+all_server_deps = all_extra_deps + ['flask']
 
 # TODO: do we want to add ALL services here?
 # i.e. even those without extra dependencies.
 # Would be good for future-compatibility, I guess.
 extras_per_service = {
-    "ec2": [_dep_cryptography, _dep_sshpubkeys_py2, _dep_sshpubkeys_py3],
     'acm': [_dep_cryptography],
-    'iam': [_dep_cryptography],
-    'cloudformation': [_dep_PyYAML, _dep_cfn_lint],
-    'cognitoidp': [_dep_python_jose, _dep_python_jose_ecdsa_pin],
     'awslambda': [_dep_docker],
     'batch': [_dep_docker],
+    'cloudformation': [_dep_PyYAML, _dep_cfn_lint],
+    'cognitoidp': [_dep_python_jose, _dep_python_jose_ecdsa_pin],
+    "ec2": [_dep_cryptography, _dep_sshpubkeys_py2, _dep_sshpubkeys_py3],
+    'iam': [_dep_cryptography],
     'iotdata': [_dep_jsondiff],
+    's3': [_dep_cryptography],
     'xray': [_dep_aws_xray_sdk],
 }
 
 extras_require = {
     'all': all_extra_deps,
-    'server': ['flask'],
+    'server': all_server_deps,
 }
 
 extras_require.update(extras_per_service)

